wordpress-mobile/WordPress-iOS
# Relocating FancyAlert over to WordpressUI.framework

### Description:
The upcoming WordPressAuth.framework requires the FancyAlert component, since it's used in severals spots during the login flow.

In this PR we're relocating this awesome component, over to WordPressUI.framework.

### Details:
- We've adopted **UIAppearance**'s API, in order to cut ties with **WPStyleGuide**.
- Since **UIAppearance** only works with UIView, we had to encapsulate all of the FancyAlertViewController's outlets within a UIView subclass.
- *Also* reloacted **FlingableViewHandler** over to **WordPressUI**
- We've cut ties with **NUXButton**, which used an Asset (and has a superclass). Now... we've got **FancyButton**, which is expected to look 99.9% like the old asset.

### Notes:
For future reference (and mostly for myself!!) the trick to get something exposed to UIAppearance is to use `@objc dynamic` modifiers, as follows:

```
@objc dynamic var someColor: UIColor
```


### Testing: Scenario A
1. Fresh Install
2. Press the `Log In` button
3. Press `Log in by entering your site address`
4. Press `Need help finding your site addres?`

Verify that the FancyAlert shows up correctly.

### Testing: Scenario B
1. Checkout Commit [8742fd1](https://github.com/wordpress-mobile/WordPress-iOS/pull/8968/commits/8742fd1990878c5033633c880e30d6b883899ece) (EY James!! thanks for this epic tip!!)
2. Log into any account
3. Open Aztec
4. _Try_ to post anything on a testing blog

Verify that the FancyAlert's Verification shows up onscreen.

--

@frosty may i bug you with a review?
(Apologies about the length of the PR!! thanks in advance!!)

H/T @diegoreymendez and @SergioEstevao for the **UIAppearance** know how.

cc @mindgraffiti
